packages/DataGrid/src/components/Cell/Cell.js

```diff
@@ -0,0 +1,72 @@
+import React from "react";
+import PropTypes from "prop-types";
+import * as styled from "../../DataGrid.styles";
+
+const propTypes = {
+  a11yText: PropTypes.string.isRequired,
+  column: PropTypes.shape({ cellProps: PropTypes.func, cell: PropTypes.oneOfType([PropTypes.string, PropTypes.func]) })
+    .isRequired,
+  columnIndex: PropTypes.number.isRequired,
+  data: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
+  gridId: PropTypes.string.isRequired,
+  rowIndex: PropTypes.number.isRequired,
+  style: PropTypes.shape({}).isRequired,
+};
+
+const Cell = React.forwardRef((props, ref) => {
+  const { style, gridId, columnIndex, rowIndex, column, data, a11yText } = props;
+  const [isActiveCell, setIsActiveCell] = React.useState(false);
+  const [isActiveRow, setIsRowActive] = React.useState(false);
+
+  React.useImperativeHandle(ref, () => ({
+    isActiveCell: isActive => {
+      setIsActiveCell(() => isActive);
+    },
+    highlightOnRow: _rowIndex => {
+      setIsRowActive(() => {
+        return Number.parseInt(_rowIndex, 10) === rowIndex;
+      });
+    },
+    deemphasizeOnrow: _rowIndex => {
+      setIsRowActive(() => {
+        return !Number.parseInt(_rowIndex, 10) === rowIndex;
+      });
+    },
+    getIndexes: () => {
+      return { rowIndex, columnIndex };
```

Comment:
nit: do these return statements all need to be on their own line?

Reply:
I prefer the return statement because it's easier to either put a console.log or a debugger statement if you use the short version would force you to modify the code each time you want to add it. even 1 line, and in the end I think this is just preference which doesn't interfere with the code itself.